### PR TITLE
ddl, stat: switch to new struct for exchange/reorg/add partition

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #55723

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/partition.go [hunk 1]
```diff
@@ -2939,12 +2939,13 @@ func (w *worker) onExchangeTablePartition(jobCtx *jobContext, t *meta.Meta, job
 	}
 
 	job.FinishTableJob(model.JobStateDone, model.StateNone, ver, pt)
-	exchangePartitionEvent := statsutil.NewExchangePartitionEvent(
-		job.SchemaID,
```
Comment (D3Hunter) [inline pkg/ddl/partition.go:2943]: this id is not needed now, how we get it now?

File: pkg/ddl/util/schema_change_notifier.go [hunk 2]
```diff
@@ -118,115 +132,210 @@ func NewDropTableEvent(
...
+	globalTableInfo *model.TableInfo,
+	droppedPartInfo *model.PartitionInfo,
+) {
 	intest.Assert(s.tp == model.ActionDropTablePartition)
-	return s.newTableInfo, s.deletedPartInfo
+	return s.tableInfo, s.droppedPartInfo
+}
+
+// NewExchangePartitionEvent creates a SchemaChangeEvent whose type is
+// ActionExchangeTablePartition.
```
Comment (fzzf678) [inline pkg/ddl/util/schema_change_notifier.go:261]:
```suggestion
// NewExchangePartitionEvent.
```

**Review by fzzf678 — APPROVED** [review]

File: pkg/statistics/handle/ddl/reorganize_partition.go [hunk 3]
```diff
@@ -15,10 +15,10 @@
 package ddl
 
 import (
-	"github.com/pingcap/tidb/pkg/statistics/handle/util"
+	ddlutil "github.com/pingcap/tidb/pkg/ddl/util"
```
Comment (D3Hunter) [inline pkg/statistics/handle/ddl/reorganize_partition.go:18]: maybe move those structs to a named pkg later, such as `schemachange` or `evnets`, `util` mostly means something more common

**Review by D3Hunter — APPROVED** [review]

File: pkg/statistics/handle/ddl/reorganize_partition.go (hunk 3 above)
Comment (lance6716) [reply pkg/statistics/handle/ddl/reorganize_partition.go:18]: Yes, will do it in future

File: pkg/ddl/util/schema_change_notifier.go [hunk 4]
```diff
@@ -28,11 +28,12 @@ import (
 // to retrieve the needed information.
 type SchemaChangeEvent struct {
 	// todo: field and method will be added in the next few pr on demand
```
Comment (Rustin170506) [inline pkg/ddl/util/schema_change_notifier.go:30]: When will we delete this TODO?

File: pkg/ddl/util/schema_change_notifier.go [hunk 5]
```diff
@@ -28,11 +28,12 @@ import (
...
-	newTableInfo    *model.TableInfo
+	tableInfo       *model.TableInfo
 	oldTableInfo    *model.TableInfo
 	addedPartInfo   *model.PartitionInfo
-	deletedPartInfo *model.PartitionInfo
+	droppedPartInfo *model.PartitionInfo
 	columnInfos     []*model.ColumnInfo
+	nonPartTableID  int64
```
Comment (Rustin170506) [inline pkg/ddl/util/schema_change_notifier.go:36]: Add comments for it.

**Review by Rustin170506 — APPROVED** [review]
Some nits

/hold

Feel free to unhold it after you address them.

File: pkg/ddl/util/schema_change_notifier.go (hunk 4 above)
Comment (lance6716) [reply pkg/ddl/util/schema_change_notifier.go:30]: There're 2 type left. Will delete it in the next PR where we finished handling all types

**Pull request merged**